src/main/java/org/testcontainers/containers/TarantoolCartridgeContainer.java

```diff
@@ -186,13 +184,20 @@ public TarantoolCartridgeContainer(String dockerFile, String buildImageName, Str
                 instancesFile, topologyConfigurationFile);
     }
 
+
     private TarantoolCartridgeContainer(Future<String> image, String instancesFile, String topologyConfigurationFile) {
         super(image);
         if (instancesFile == null || instancesFile.isEmpty()) {
             throw new IllegalArgumentException("Instance file name must not be null or empty");
         }
+        if (topologyConfigurationFile == null || topologyConfigurationFile.isEmpty()) {
+            throw new IllegalArgumentException("Topology configuration file must not be null or empty");
+        }
+        String instances = topologyConfigurationFile.substring(topologyConfigurationFile.lastIndexOf('.') + 1);
+        if (instances.equals("lua")) {
```

Comment:
> Why not to set topologyConfigurationFile to yml ?

Because the user can transfer both yml and lua file whith topology to topologyConfigurationFile . And strategy we checkout  depends type a file.

> Maybe move this 'if' to setupTopology() ?
Because if we get lua file we must initialize the variable topologyConfigurationFile immediately in the constructor